### PR TITLE
feat: cache preprocessed datasets for higher performance

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Joblib's cache uses the inputs for cache invalidation - would especially like comments on whether this seems reasonable. Don't want invalidation errors again :-(

### CODE REVIEW

**Review by HLasse — APPROVED** [review]
See below

File: src/psycop_model_training/data_loader/utils.py [hunk 1]
```diff
@@ -13,32 +19,43 @@
...
 
+memory_path = Path("E:/shared_resources/feature_cache")
+
+# Delete any file older than 24 hours in memory_path
+for file in memory_path.glob("*"):
+    one_day = time.time() - 24 * 3600
+    if file.stat().st_mtime < one_day:
+        file.unlink()
+
+memory = Memory(location=memory_path, verbose=0)
```
Comment (HLasse) [inline src/psycop_model_training/data_loader/utils.py:30]: This is gonna break if you install the module locally. Perhaps move the `memory_path` to the config and run the deletion before `load_and_filter_split_from_cfg` is called?

**Review by HLasse — CHANGES_REQUESTED** [review]

File: src/psycop_model_training/data_loader/utils.py (hunk 1 above)
Comment (MartinBernstorff) [reply src/psycop_model_training/data_loader/utils.py:30]: You're right - and the tests are failing as well. Will look at it, thanks!

**Comment by MartinBernstorff** [comment]
I've thought about using the current git commit as a hash, but that also seems overly strict. Not quite sure how to do this effectively, so might end up just not doing it at all :-(

**Pull request closed without merging**